ducktape/tests/reporter.py

```diff
@@ -12,98 +12,131 @@
+            "status:     %s" % pass_fail(self.result.success),
+            "run time:   %s" % format_time(self.result.run_time),
+        ]
+
+        if self.result.data is not None:
+            result_lines.append(json.dumps(self.result.data))
+
+        return "\n".join(result_lines)
+
+    def report_string(self):
+        """Get the whole report string."""
+        return "\n".join(["=" * self.width,
+                          self.result_string()])
 
-        r = ""
-        if minutes > 0:
-            r += "%d minute%s " % (minutes, "" if minutes == 1 else "s")
-        r += "%.3f seconds" % seconds
-        return r
 
+class SingleResultStdoutReporter(SingleResultReporter):
     def report(self):
-        raise NotImplementedError("method report must be implemented by subclasses of TestReporter")
+        print self.report_string()
+
+
+class SingleResultFileReporter(SingleResultReporter):
+    def report(self):
+        self.width = DEFAULT_SEPARATOR_WIDTH
+        report_file = os.path.join(self.result.test_context.results_dir, "report.txt")
+        with open(report_file, "w") as fp:
+            fp.write(self.report_string())
+
+        # write collected data
+        if self.result.data is not None and len(self.result.data) > 0:
+            data_file = os.path.join(self.result.test_context.results_dir, "data.json")
+            with open(data_file, "w") as fp:
+                fp.write(json.dumps(self.result.data))
+
+
+class SummaryReporter(object):
+    def __init__(self, results):
+        self.results = results
+        self.width = get_terminal_size()[0]
+
+    def report(self):
+        raise NotImplementedError("method report must be implemented by subclasses of SummaryReporter")
 
 
-class SimpleReporter(TestReporter):
+class SimpleSummaryReporter(SummaryReporter):
     def header_string(self):
         """Header lines of the report"""
         header_lines = [
-            "=========================================================================================",
+            "=" * self.width,
+            "SESSION REPORT (ALL TESTS)",
             "session_id: %s" % self.results.session_context.session_id,
-            "run time:   %s" % self.format_time(self.results.run_time),
+            "run time:   %s" % format_time(self.results.run_time),
             "tests run:  %d" % len(self.results),
             "passed:     %d" % self.results.num_passed(),
             "failed:     %d" % self.results.num_failed(),
-            "========================================================================================="
+            "=" * self.width
         ]
 
         return "\n".join(header_lines)
 
-    def result_string(self, result):
-        """Stringify a single result."""
-
-        result_lines = [
-            self.pass_fail(result.success) + ":     " + result.test_name,
-            "run time: %s" % self.format_time(result.run_time)
-            ]
-
-        if not result.success:
```

Comment:
As far as I can tell, there's no longer an equivalent to this code which would log the summary info when there's an error. So I think we now have no logs containing the actual error.